libs/common/src/platform/abstractions/config/config.service.abstraction.ts

```diff
@@ -6,8 +6,7 @@ import { ServerConfig } from "./server-config";
 
 export abstract class ConfigServiceAbstraction {
   serverConfig$: Observable<ServerConfig | null>;
-  fetchServerConfig: () => Promise<ServerConfig>;
-  getFeatureFlagBool: (key: FeatureFlag, defaultValue?: boolean) => Promise<boolean>;
-  getFeatureFlagString: (key: FeatureFlag, defaultValue?: string) => Promise<string>;
-  getFeatureFlagNumber: (key: FeatureFlag, defaultValue?: number) => Promise<number>;
+  getFeatureFlag$: <T>(key: FeatureFlag, defaultValue?: T) => Observable<T>;
+  getFeatureFlag: <T>(key: FeatureFlag, defaultValue?: T) => Promise<T>;
```

Comment:
I'd like to see some type restrictions on `T`
```suggestion
  getFeatureFlag$: <T extends boolean | number | string>(key: FeatureFlag, defaultValue?: T) => Observable<T>;
  getFeatureFlag: <T extends boolean | number | string>(key: FeatureFlag, defaultValue?: T) => Promise<T>;
```

Reply:
Great suggestion! done